container_pipeline/cleanup_registry/garbagecollector.py

```diff
@@ -10,53 +12,118 @@ class GarbageCollector(object):
         """
         Initialize the garbage collector object.
 
         :param registry_host: The ip or host name of registry to query. Default is 127.0.0.1.
         :param registry_port: The port of the registry to query. Default is 5000
         :param registry_secure: Is the registry secure or insecure (https or http)
-        :param local_index: Is the index to query locally available.
+        :param local_index: Is the index to query locally available. This means no cloning.
         :param index_git: If the index is on a git repo, the url of that repo.
+        :param index_location: The path where the index is available or needs to be cloned.
         :param verbose: If set then steps are displayed on screen as they happen.
         :param collect: If set. then the garbage collection takes place, else it does dry run.
         """
+
         self._verbose = verbose
-        self._index_location = "./c_i"
+        self._index_location = index_location
         self._collect = collect
-        if local_index:
-            # If local index is set, check if inex files are present at expected location.
-            if not lib.path.exists(self._index_location):
-                raise Exception("Local index specified, but does not exist")
-        else:
-            # Otherwise clone index.
-            if self._verbose:
-                print("Cloning container index...")
-            lib.clone_repo(index_git, self._index_location)
+        self._local_index = local_index
         self._index_git = index_git
+        self._gc_exceptions = []
+        self._gc_match_only = []
         # Formulate the registry url
         self._registry_url = str.format("{schema}://{host}{port}/v2",
                                         schema="https" if registry_secure else "http",
                                         host=registry_host,
                                         port="" if not registry_port else ":" + registry_port
                                         )
-        self._index_location = self._index_location + "/index.d"
+        self._index_d_location = self._index_location + "/index.d"
         # Setup reg info object to query the registry and cache metadata.
+        self._index_containers = {}
+        self._mismatched = {}
+
+    def _query_registry(self):
         if self._verbose:
             print("Gathering registry metadata...")
         self._registry_info = lib.RegistryInfo(self._registry_url)
-        self.index_containers = {}
-        self.mismatched = {}
 
-    def _delete_mismatched(self):
+    def _query_index(self):
+        """
+        Query a cloned or local container index.
+        :return:
+        """
+        index_files = glob(self._index_d_location + "/*.yml")
+        for index_file in index_files:
+            if "index_template" not in index_file:
+                data = lib.load_yaml(index_file)
+                if "Projects" not in data:
+                    raise Exception("Invalid index file")
```

Comment:
should this log issue for file in process and continue processing rest of the files in index ?

Reply:
@navidshaikh no, if an index file is wrong and we allow it to proceed, we might end up with incorrect images being deleted.